Libplanet.Explorer/Queries/BlockQuery.cs

```diff
@@ -43,13 +43,30 @@ public BlockQuery()
             Field<BlockType<T>>(
                 "block",
                 arguments: new QueryArguments(
-                    new QueryArgument<IdGraphType> { Name = "hash" }
+                    new QueryArgument<IdGraphType> { Name = "hash" },
+                    new QueryArgument<IdGraphType> { Name = "index" }
                 ),
                 resolve: context =>
                 {
-                    HashDigest<SHA256> hash = HashDigest<SHA256>.FromString(
-                        context.GetArgument<string>("hash"));
-                    return Query<T>.GetBlock(hash);
+                    string? hash = context.GetArgument<string?>("hash", null);
+                    long? index = context.GetArgument<long?>("index", null);
+
+                    if ((hash is string && index is long) || (hash is null && index is null))
+                    {
+                        throw new System.Exception("Unexpected block query");
```

Comment:
How about using `GraphQL.ExecutionError` instead?

You can see https://graphql-dotnet.github.io/docs/getting-started/errors/ for detail.